docker/Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM ubuntu:xenial
+
+RUN apt-get update && apt-get install -y \
+        curl \
+        lsb-release \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN curl -sS https://dl.google.com/cloudagents/install-logging-agent.sh | bash
+
+RUN /opt/google-fluentd/embedded/bin/gem uninstall fluent-plugin-google-cloud \
+    && /opt/google-fluentd/embedded/bin/gem install fluent-plugin-google-cloud -v 0.6.4.pre.3 \
+    && apt-get autoremove
+
+COPY google-fluentd.conf /etc/google-fluentd/google-fluentd.conf
+
+CMD sed -i "s/metadata-agent/$METADATA_AGENT_HOSTNAME/" "/etc/google-fluentd/google-fluentd.conf" \
```

Comment:
Even if the value were defaulted to "", it would make it very clear that we have some sort of dependency on an environment variable called "METADATA_AGENT_HOSTNAME". It's ok if we don't go down this path, but the sed substitution might look a bit magical without a defined environment variable, regardless of the default value. Alternatively, we could just add a comment that offers a few suggestions for which values are reasonable under the various scenarios you just mentioned.